lib/glific/flows.ex

```diff
@@ -571,13 +571,15 @@ defmodule Glific.Flows do
   @status "published"
 
   @doc """
-  Start flow for a contact
+  Start flow for a contact and cache the result
   """
   @spec start_contact_flow(Flow.t() | integer, Contact.t()) ::
           {:ok, Flow.t()} | {:error, String.t()}
   def start_contact_flow(flow_id, %Contact{} = contact) when is_integer(flow_id) do
-    {:ok, flow} = get_cached_flow(contact.organization_id, {:flow_id, flow_id, @status})
-    process_contact_flow([contact], flow, @status)
+    case get_cached_flow(contact.organization_id, {:flow_id, flow_id, @status}) do
+      {:ok, flow} -> process_contact_flow([contact], flow, @status)
+      {:error, error} -> {:error, error}
```

Comment:
@siddhant3030 can you check if a test case is there or add this in the test case?